cmd/playbook.go

```diff
@@ -0,0 +1,77 @@
+package cmd
+
+import (
+	"github.com/mitchellh/cli"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+type PlaybookInterface interface {
+	SetFiles(files map[string]string)
+	SetRoot(root string)
+	Run(playbookYml string, args []string, ui cli.Ui) error
+}
+
+type Playbook struct {
+	files map[string]string
+	root  string
+}
+
+func (p *Playbook) SetFiles(files map[string]string) {
```

Comment:
🤔 I'm not sure how I feel about `Run` optionally doing the generation of playbook templates. It's a bit too implicit and inconsistent? Since in other cases it's used to run "normal" playbooks which already exist (and won't be removed).

Couple options:
1. leave as is
2. rename `SetFiles` to be `GeneratePlaybooks` or something more explicit
3. add a new version of `Run` such as `RunTemporary` to make it more explicit

Reply:
Seperated `AdHocPlaybook` & `Playbook`